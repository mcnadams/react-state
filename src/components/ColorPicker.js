import React from 'react';
import PropTypes from 'prop-types';
import styles from './Color.css';

function ColorPicker({ colorUpdater }) {
  return (
    <>
      <button className={styles.red} onClick={() => colorUpdater('red')}>Red</button>
      <button className={styles.yellow} onClick={() => colorUpdater('yellow')}>Yellow</button>
      <button className={styles.blue} onClick={() => colorUpdater('blue')}>Blue</button>
    </>
  );
}

ColorPicker.propTypes = {
  colorUpdater: PropTypes.func.isRequired
};

export default ColorPicker;
