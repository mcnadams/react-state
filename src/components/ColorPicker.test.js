import React from 'react';
import { shallow } from 'enzyme';
import ColorPicker from './ColorPicker';

describe('ColorPicker', () => {
  it('renders a ColorPicker', () => {
    const wrapper = shallow(<ColorPicker colorUpdater={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('selects red', () => {
    const colorUpdater = jest.fn();
    const wrapper = shallow(<ColorPicker colorUpdater={colorUpdater} />);
    wrapper.find('button').at(0).simulate('click');
    expect(colorUpdater).toHaveBeenCalledWith('red');
  });

  it('selects yellow', () => {
    const colorUpdater = jest.fn();
    const wrapper = shallow(<ColorPicker colorUpdater={colorUpdater} />);
    wrapper.find('button').at(1).simulate('click');
    expect(colorUpdater).toHaveBeenCalledWith('yellow');
  });

  it('selects blue', () => {
    const colorUpdater = jest.fn();
    const wrapper = shallow(<ColorPicker colorUpdater={colorUpdater} />);
    wrapper.find('button').at(2).simulate('click');
    expect(colorUpdater).toHaveBeenCalledWith('blue');
  });

});
