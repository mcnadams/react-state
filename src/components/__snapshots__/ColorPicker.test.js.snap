// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Picker renders a Color Button 1`] = `
<Fragment>
  <button
    className="red"
    onClick={[Function]}
  >
    Red
  </button>
  <button
    className="yellow"
    onClick={[Function]}
  >
    Yellow
  </button>
  <button
    className="blue"
    onClick={[Function]}
  >
    Blue
  </button>
  <div
    className=""
    id="color-block"
  />
</Fragment>
`;
